Common/Utility/DificultyUtils.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Runtime.CompilerServices;
+using Terraria;
+using Terraria.ModLoader;
+
+namespace TheBereftSouls.Common.Utility
+{
+    public class DificultyUtils : ModSystem
+    {
+        public static bool Revengeance = false;
+        public static bool Death = false;
+        public static bool EternityMode = false;
+        public static bool MasochistMode = false;
+        public static bool LegendaryMode = false;
+
+        public override void PostUpdateWorld()
+        {
+            if (TheBereftSouls.CalamityMod != null && TheBereftSouls.FargosSoulMod != null)
+            {
+                CalamityDificultyCheck.CheckCalamity(ref Revengeance, ref Death);
+                FargosDificultyCheck.CheckFargos(ref EternityMode, ref MasochistMode);
+            }
+            else if (TheBereftSouls.CalamityMod != null)
+            {
+                CalamityDificultyCheck.CheckCalamity(ref Revengeance, ref Death);
+            }
+            else if (TheBereftSouls.FargosSoulMod != null)
+            {
+                FargosDificultyCheck.CheckFargos(ref EternityMode, ref MasochistMode);
+            }
+            LegendaryMode = Main.masterMode && Main.getGoodWorld;
+
+            base.PostUpdateWorld();
```

Comment:
No `base.PostUpdateWorld()` functionality to call

https://github.com/tModLoader/tModLoader/blob/2569fc2ce3c862697034cd91e455c769f9f55517/patches/tModLoader/Terraria/ModLoader/ModSystem.cs#L211